lib/retry.ts

```diff
@@ -91,6 +91,20 @@ function validateRetryConfig<T>(config: RetryConfig<T>): void {
   }
 }
 
+
+async function checkInternet(): Promise<boolean> {
+  return new Promise((resolve) => {
+    require("dns").lookup("ms.portal.azure.com", function (err: any): void {
+      if (err && err.code === "ENOTFOUND") {
```

Comment:
Would it be possible that the this dns query resolves even when the connection is not there? For instance,  if there is a dns cache it might be possible that the value is returned from the cache without doing a network call.

Reply:
Good catch!
You are right, the `lookup` method indeed reuses the dns cache and therefore not guaranteed to perform any network communication. Therefore, we will have to use the `resolve` method instead of `lookup` which will query the dns server and hence always use the network to perform queries.

Reply:
It's still possible to "resolve" without having "internet" FWIW. This occurs often on our wifi network - DNS queries work as the DNS server is within our LAN (still a few hops away), but access to the outside world is broken. I think this feature should be understood as a 90% sort of solution for "ethernet unplugged" and "switching to/from wifi" scenarios.